Prose/ProseLib/Sources/TestHelpers/UUID+Incrementing.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+public extension UUID {
+    /// A deterministic, auto-incrementing "UUID" generator for testing.
+    static var incrementing: () -> UUID {
+        var uuid = 0
+        return {
+            defer { uuid += 1 }
+            return UUID(uuidString: "00000000-0000-0000-0000-\(String(format: "%012x", uuid))")!
```

Comment:
I'm not sure this is a valid UUIDv4, but `UUID` doesn't validate it IIRC, so it's OK.